share/spice/goodreads/goodreads.js

```diff
@@ -0,0 +1,83 @@
+(function(env) {
+  "use strict";
+
+  env.ddg_spice_goodreads = function(api_result) {
+    console.info(api_result);
+
+    // Validating response
+    if(!api_result || api_result.GoodreadsResponse.search["total-results"].text == "0") {
+      return Spice.failed('goodreads');
+    }
+        
+    var get_details_uri = '/js/spice/goodreads/get_book_details/',
+      get_image_for_book_uri = '/js/spice/goodreads/get_image_for_book/';
+
+    $.ajaxSetup({ cache: true });
+    
+    // Render the response
+    Spice.add({
+      id: "goodreads",
+      name: "Books",
+      data: api_result.GoodreadsResponse.search.results.work,
+      meta: {
+        primaryText: "Book results for " + api_result.GoodreadsResponse.search.query.text,
+        sourceName: "GoodReads",
+        sourceUrl: "https://www.goodreads.com/"
+      },
+      
+      
+      onItemShown: function(item) {  
+        // Good reads description API
+        var book = item.best_book;
+  
+        // *** This ajax call is not working ***
+        $.getJSON('/js/spice/goodreads/get_book_details/' + book.id.text, function(data) {
```

Comment:
I'm looks like the Goodreads API is ratelimiting us after we make the first few API calls. I'm not sure if that's he case though but the details calls start dying pretty quickly. Please investigate what's going on. 

Once that's fixed it looks like this is just about done!